custom_components/candy/client/model.py

```diff
@@ -181,28 +185,279 @@ class OvenState(StatusCode):
+    heater2: Optional[HobHeaterStatus]
+    heater3: Optional[HobHeaterStatus]
+    heater4: Optional[HobHeaterStatus]
+    lock: bool
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=HobStatus.parse_state(json),
+            heater1=HobHeaterStatus.from_json(json, "1")
+            if "Z1status" in json
+            else None,
+            heater2=HobHeaterStatus.from_json(json, "2")
+            if "Z2status" in json
+            else None,
+            heater3=HobHeaterStatus.from_json(json, "3")
+            if "Z3status" in json
+            else None,
+            heater4=HobHeaterStatus.from_json(json, "4")
+            if "Z4status" in json
+            else None,
+            lock=json["lock"] == "1",
+            remote_control=json["StatoWiFi"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Hob status
+        """
+        if (
+            json["Z1status"] == "1"
+            or json["Z2status"] == "1"
+            or json["Z3status"] == "1"
+            or json["Z4status"] == "1"
+        ):
+            return HobState.HEATING
+
+        if (
+            json["Z1hot"] == "1"
+            or json["Z2hot"] == "1"
+            or json["Z3hot"] == "1"
+            or json["Z4hot"] == "1"
+        ):
+            return HobState.COOLING_DOWN
+        else:
+            return HobState.IDLE
+
+
+class HoodState(StatusCode):
+    IDLE = (0, "Idle")
+    FAN = (1, "Fan")
+    LIGHT = (2, "Light")
+    FAN_LIGHT = (3, "Fan & Light")
+
+
+@dataclass
+class HoodStatus:
+    machine_state: HoodState
+    light: bool
+    fan: bool
+    grease_filter: bool
+    carbon_Filter: bool
+    warning: bool
+    remote_control: bool
+
+    @classmethod
+    def from_json(cls, json):
+        return cls(
+            machine_state=HoodStatus.parse_state(json),
+            light=json["Light"] == "1",
+            fan=int(json["Fan"]) > 0,
+            grease_filter=json["GreaseFilter"] == "1",
+            carbon_Filter=json["CarbonFilter"] == "1",
+            warning=json["Warning"] == "1",
+            remote_control=json["WiFiStatus"] == "1",
+        )
+
+    @staticmethod
+    def parse_state(json) -> str:
+        """
+        Parse Hood status
+        """
+        if int(json["Fan"]) > 0 and json["Light"] == "1":
+            return HoodState.FAN_LIGHT
+
+        if int(json["Fan"]) > 0:
+            return HoodState.FAN
+
+        if json["Light"] == "1":
+            return HoodState.LIGHT
+        else:
+            return HoodState.IDLE
+
+
+class FridgeState(StatusCode):
+    NORMAL = (0, "Normal")
+    ECO = (1, "Eco")
+    SUPER_FREEZING = (2, "Super freezing")
+    SMART_COOLING = (3, "Smart cooling")
+    SUPER_FREEZING_COOLING = (4, "Super freezing & cooling")
+    ERROR = (5, "Error")
+
+
+@dataclass
+class FridgeStatus:
+    machine_state: HobState
+    cooling_temperature: bool
```

Comment:
What is the meaning of this boolean field exactly? It's not clear for me from the variable name

Reply:
It is actualy integer. I change it into integer. It show fridge cooling zone temperature and freezing zone temperature.